server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java

```diff
@@ -225,42 +226,46 @@ protected Object parseSourceValue(Object value) {
         @Override
         public Query termQuery(Object value, @Nullable QueryShardContext context) {
             failIfNotIndexedAndNoDocValues();
-            Query query;
-            if (value instanceof InetAddress) {
-                query = InetAddressPoint.newExactQuery(name(), (InetAddress) value);
-            } else {
-                if (value instanceof BytesRef) {
-                    value = ((BytesRef) value).utf8ToString();
-                }
-                String term = value.toString();
-                if (term.contains("/")) {
-                    final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
-                    query = InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
+            final PointRangeQuery pointQuery;
+            {
```

Comment:
Sorry why we use `{` block here?

``
            final Query query;
            if (value instanceof InetAddress) {
                query = InetAddressPoint.newExactQuery(name(), (InetAddress) value);
            } else {
                if (value instanceof BytesRef) {
                    value = ((BytesRef) value).utf8ToString();
                }
                String term = value.toString();
                if (term.contains("/")) {
                    final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
                    query = InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
                } else {
                    InetAddress address = InetAddresses.forString(term);
                    query = InetAddressPoint.newExactQuery(name(), address);
                }
            }

            final PointRangeQuery pointQuery = (PointRangeQuery) query;
            ...
```

Reply:
It encloses `query`, exposing only `final pointQuery` for later usage. It's little bit boring, for sure. I can drop it if you wish.

Reply:
The point query could be assigned from the  `query` itself (as per snippet)

Reply:
ahhh right. Did it. Pushed. Thanks.